bin/process-csv.js

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env node
+/**
+ * process-csv.js
+ * Converts source database derived CSV data into CSV data compatible with the Magento2 importer.
+ */
+(function (_, fs, csv, slug) {
+    'use strict';
+    var categoryMap, processProducts;
+
+    // Set RFC mode for slug generation.
+    slug.defaults.mode = 'rfc3986';
+
+    // Map for category values
+    categoryMap = {
+        'AUTOBIOGRAPHY / BIOGRAPHY': 'Default Category/Books/Biographical',
+        'EXPLORATION AND TRAVEL': 'Default Category/Books/Travel',
+        'GOLD': 'Default Category/Books/Gold',
+        'HISTORICAL NOVEL': 'Default Category/Books/Historical Novel',
+        'INDIGENOUS HISTORY   [ABORIGINES]': 'Default Category/Books/Indigenous History',
+        'INDIGENOUS HISTORY (Aborigines)': 'Default Category/Books/Indigenous History',
+        'LOCAL AND REGIONAL HISTORY': 'Default Category/Books/Local History',
+        'MARITIME HISTORY': 'Default Category/Books/Maritime',
+        'MILITARY': 'Default Category/Books/Military',
+        'RAILWAYS AND TRANSPORT': 'Default Category/Books/Railways & Transport',
+        'SOCIAL HISTORY': 'Default Category/Books/Social History',
+        'STATEWIDE': 'Default Category/Books/State History',
+        'STUDIES IN WESTERN AUSTRALIAN HISTORY': 'Default Category/Books/State History',
+        'WA HISTORY - STATEWIDE': 'Default Category/Books/State History',
+        'YOUTH AND CHILDREN': 'Default Category/Books/Youth & Children'
+    };
+
+    // Generate "products.csv" based on the source data.
+    processProducts = function (workPath, targetPath) {
+        console.log('Processing CSV data with workPath="%s", targetPath="%s"', workPath, targetPath);
+        fs.readdir(targetPath + '/images', function (err, imageFilenames) {
+            if (err) {
+                console.error('Could not read images path "%s", please check the path exists and try again', targetPath + '/images');
+                return;
+            }
+            fs.createReadStream(workPath + '/Books.csv')
+                .pipe(csv.parse({ columns: true }))
+                .pipe(csv.transform(function (record) {
+                    var title, imageFilename;
+
+                    if (!record.RRP) {
```

Comment:
OK I've added a reporting mechanism for skipped records, on the line following this one (47) and from line 149 actually creates the `skipped-records.csv` file.  Apparently I can't attach CSV files to comments but here's the contents, it's only 12 records:

```
#,BookCode,Title,Reason for Skipping
1,545,Cossack Gold: The Romance of the Kimberleys  DO NOT USE,RRP is zero or missing
2,1201,"Journal of the Historical Society of South Australia - Number 41, 2013",RRP is zero or missing
3,1206,use this space,RRP is zero or missing
4,1228,use this space 1,RRP is zero or missing
5,1239,Vanishing Village,RRP is zero or missing
6,1256,use this space 3,RRP is zero or missing
7,1257,Banboo Curtain over Port Hedland,RRP is zero or missing
8,1300,Secret No Longer,RRP is zero or missing
9,1347,Old Fremantle Childhood,RRP is zero or missing
10,1429,C,RRP is zero or missing
11,1461,B,RRP is zero or missing
12,1463,Carala Zampatti,RRP is zero or missing
```